MapboxNavigation/StyleManager.swift

```diff
@@ -1,49 +1,63 @@
 import UIKit
 import Solar
+import MapboxCoreNavigation
 
 /**
  The `StyleManagerDelegate` protocol defines a set of methods used for controlling the style.
  */
-@objc(MBStyleManagerDelegate)
-public protocol StyleManagerDelegate: NSObjectProtocol {
+public protocol StyleManagerDelegate: class, UnimplementedLogging {
     /**
-     Asks the delegate for a location to use when calculating sunset and sunrise.
+     Asks the delegate for a location to use when calculating sunset and sunrise
+     - note: This delegate method includes a default implementation that prints a warning to the console when this method is called. See `UnimplementedLogging` for details.     .
      */
-    @objc(locationForStyleManager:)
     func location(for styleManager: StyleManager) -> CLLocation?
     
     /**
      Informs the delegate that a style was applied.
+     - note: This delegate method includes a default implementation that prints a warning to the console when this method is called. See `UnimplementedLogging` for details.
      
-     This delegate method is the equivalent of `Notification.Name.styleManagerDidApplyStyle`.
+     - note: This delegate method is the equivalent of `Notification.Name.styleManagerDidApplyStyle`.
```

Comment:
`note`s are pretty loud. Can this just be a normal paragraph before the other `note`?